### ISSUE TITLE
StaticSite: Allow the creation of Route53 records to be disabled

### ISSUE THREAD

**Issue by mtdewers** [serverless-stack/serverless-stack#598]
In our organization, Route53 is managed from a central AWS account for which we do not have permission.  It would be great if StaticSite allowed me to disable the Route53 record creation.

**Comment by fwang** [comment]
Hey @mtdewers, currently the Route53 record only gets created if the `customDomain` is specified. Does that work for you?

Or are you looking to set the `customDomain`, but don't want to create the Route53 record?

**Comment by mtdewers** [comment]
Looking to be able to use `customDomain ` but not create the Route53 record.  Like that the construct takes care of the cert, cloud front, etc.  but  in our environment we can't allow it to do the Route53 changes.  Even something as simple as making the code that performs this work overridable in a class that extends `StaticSite `would be a big help (it currently is marked private).

**Comment by fwang** [comment]
@mtdewers are you going to manually create the Route53 change? Without it, the custom domain won't work.

**Comment by mtdewers** [comment]
Yes.  We are forced to by company policy to route all Route53 changes to a central group. I know this is likely a bit of an edge case which is why it would acceptable to me if all the logic surrounding the Route53 work were contained in a protected method on StaticSite that I could override in a class that extends it.  In fact, you might consider not using private methods in the code base at all.

**Comment by fwang** [comment]
@mtdewers I restructured the code a bit and made the `createRoute53Records()` protected. You can override it like this:
```
  class MyStaticSite extends StaticSite {

    protected createRoute53Records(): void {
      // leave this blank to not have Route 53 records created
    }
  }
```

I will cut a release later today.

^^ btw I like your idea of making the code base more inheritable. I'm going to expose more private methods to protected over time.

**Comment by fwang** [comment]
Released in [v0.35.2](https://github.com/serverless-stack/serverless-stack/releases/tag/v0.35.2)
